trade-module/src/main/java/io/edpn/backend/trade/adapter/config/ServiceConfig.java

```diff
@@ -0,0 +1,204 @@
+import io.edpn.backend.trade.application.service.RequestStationArrivalDistanceService;
+import io.edpn.backend.trade.application.service.RequestStationLandingPadSizeService;
+import io.edpn.backend.trade.application.service.RequestStationPlanetaryService;
+import io.edpn.backend.trade.application.service.RequestStationRequireOdysseyService;
+import io.edpn.backend.trade.application.service.RequestSystemCoordinatesService;
+import io.edpn.backend.trade.application.service.RequestSystemEliteIdService;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.retry.support.RetryTemplate;
+
+import java.util.List;
+import java.util.concurrent.Executor;
+
+@Configuration("TradeServiceConfig")
+public class ServiceConfig {
+
+    @Bean(name = "tradeFindCommodityService")
+    public FindCommodityService findCommodityService(
+            LoadAllValidatedCommodityPort loadAllValidatedCommodityPort,
+            LoadValidatedCommodityByNamePort loadValidatedCommodityByNamePort,
+            LoadValidatedCommodityByFilterPort loadValidatedCommodityByFilterPort,
+            ValidatedCommodityDtoMapper validatedCommodityDTOMapper,
+            FindCommodityFilterDtoMapper findCommodityFilterDtoMapper) {
+        return new FindCommodityService(loadAllValidatedCommodityPort, loadValidatedCommodityByNamePort, loadValidatedCommodityByFilterPort, validatedCommodityDTOMapper, findCommodityFilterDtoMapper);
+    }
+
+    @Bean(name = "tradeFindCommodityMarketInfoService")
+    public FindCommodityMarketInfoService findCommodityMarketInfoService(
+            GetFullCommodityMarketInfoPort commodityMarketInfoPort,
+            CommodityMarketInfoDtoMapper commodityMarketInfoDtoMapper) {
+        return new FindCommodityMarketInfoService(commodityMarketInfoPort, commodityMarketInfoDtoMapper);
+    }
+
+    @Bean(name = "tradeLocateCommodityService")
+    public LocateCommodityService locateCommodityService(
+            LocateCommodityByFilterPort locateCommodityByFilterPort,
+            LocateCommodityFilterDtoMapper locateCommodityFilterDtoMapper,
+            LocateCommodityDtoMapper locateCommodityDtoMapper) {
+        return new LocateCommodityService(locateCommodityByFilterPort, locateCommodityFilterDtoMapper, locateCommodityDtoMapper);
+    }
+
+    @Bean(name = "tradeRecieveCommodityMessageUsecase")
+    public ReceiveCommodityMessageService receiveCommodityMessageService(
+            ExistsByStationNameAndSystemNameAndTimestampPort existsByStationNameAndSystemNameAndTimestamp,
+            LoadOrCreateSystemByNamePort loadOrCreateSystemByNamePort,
+            LoadOrCreateBySystemAndStationNamePort loadOrCreateBySystemAndStationNamePort,
+            LoadOrCreateCommodityByNamePort loadOrCreateCommodityByNamePort,
+            UpdateStationPort updateStationPort,
```

Comment:
I know, hex architecture, but... I find the large number of dependencies/parameters a bit distasteful. Would lumping some of them together into larger units with a super interface be horribly wrong?

Reply:
Agree with the distastefulness of tons of ports, but this is the standard hexagonal formatting. One port for one action. This separates your unit testing and allows flexibility of deployment.
If you bundle them you HAVE to use all of them or none of them in your adapter.